tests/codegen_testcases/solidity/scale.sol

```diff
@@ -20,38 +20,76 @@ contract ExternalFunctions {
-        // CHECK: writebuffer buffer:%abi_encoded.temp.9 offset:uint32 0 value:uint8((%temp.10 * uint32 4))
-        // CHECK: ty:uint32 %temp.11 = uint32 1
-        // CHECK: writebuffer buffer:%abi_encoded.temp.9 offset:uint32 0 value:uint16(((%temp.10 * uint32 4) | uint32 1))
-        // CHECK: ty:uint32 %temp.11 = uint32 2
-        // CHECK: writebuffer buffer:%abi_encoded.temp.9 offset:uint32 0 value:((%temp.10 * uint32 4) | uint32 2)
-        // CHECK: ty:uint32 %temp.11 = uint32 4
-        // CHECK: memcpy src: (arg #0), dest: (advance ptr: %abi_encoded.temp.9, by: (uint32 0 + %temp.11)), bytes_len: %temp.10
-        // CHECK: branchcond (unsigned more %temp.10 > uint32 63), block10, block8
+        
+    // CHECK: branchcond (unsigned more (builtin ArrayLength ((arg #0))) > uint32 1073741823), block6, block7
+
+    // CHECK: block1: # small
+    // CHECK: ty:uint32 %temp.9 = uint32 1
+    // CHECK: branch block5
+
+    // CHECK: block2: # medium
+    // CHECK: ty:uint32 %temp.9 = uint32 2
+    // CHECK: branch block5
+
+    // CHECK: block3: # medium_or_big
+    // CHECK: branchcond (unsigned more (builtin ArrayLength ((arg #0))) > uint32 16383), block4, block2
+
+    // CHECK: block4: # big
+    // CHECK: ty:uint32 %temp.9 = uint32 4
+    // CHECK: branch block5
+
+    // CHECK: block5: # done
+    // CHECK: ty:bytes %abi_encoded.temp.10 = (alloc bytes len (%temp.9 + (builtin ArrayLength ((arg #0)))))
+    // CHECK: ty:uint32 %temp.11 = (builtin ArrayLength ((arg #0)))
+    // CHECK: branchcond (unsigned more %temp.11 > uint32 1073741823), block13, block14
+
+    // CHECK: block6: # fail
+    // CHECK: assert-failure
+
+    // CHECK: block7: # prepare
+    // CHECK: branchcond (unsigned more (builtin ArrayLength ((arg #0))) > uint32 63), block3, block1
+
+    // CHECK: block8: # small
+    // CHECK: writebuffer buffer:%abi_encoded.temp.10 offset:uint32 0 value:uint8((%temp.11 * uint32 4))
+    // CHECK: ty:uint32 %temp.12 = uint32 1
+    // CHECK: branch block12
+
+    // CHECK: block9: # medium
+    // CHECK: writebuffer buffer:%abi_encoded.temp.10 offset:uint32 0 value:uint16(((%temp.11 * uint32 4) | uint32 1))
+    // CHECK: ty:uint32 %temp.12 = uint32 2
+    // CHECK: branch block12
+
+    // CHECK: block10: # medium_or_big
+    // CHECK: branchcond (unsigned more %temp.11 > uint32 16383), block11, block9
+
+    // CHECK: block11: # big
+    // CHECK: writebuffer buffer:%abi_encoded.temp.10 offset:uint32 0 value:((%temp.11 * uint32 4) | uint32 2)
+    // CHECK: ty:uint32 %temp.12 = uint32 4
+    // CHECK: branch block12
+
+    // CHECK: block12: # done
+    // CHECK: memcpy src: (arg #0), dest: (advance ptr: %abi_encoded.temp.10, by: (uint32 0 + %temp.12)), bytes_len: %temp.11
+    // CHECK: return 
+
+    // CHECK: block13: # fail
+    // CHECK: assert-failure
+
+    // CHECK: block14: # prepare
+    // CHECK: branchcond (unsigned more %temp.11 > uint32 63), block10, block8
+
```

Comment:
No tests for the decoder?
At least a test for the part that is not common with bosh is welcome.

Reply:
It is only decoding of compact integers and external functions which are covered elsewhere. But yeah it doesn't hurt to add a codegen test case for them too :+1: